### DISCUSSION TITLE
Ripemd160 and base58 encoding

### DISCUSSION THREAD

**Discussion by blmalone** [krzyzanowskim/CryptoSwift#941] [General]
**Is your feature request related to a problem? Please describe.**
To create any Bitcoin addresses, you need both of these functions. 

**Describe the solution you'd like**
Simply add a well respected implementation of the above algorithms.

**Describe alternatives you've considered**
Manually bringing an implementation in my project. Would be nice if it was packaged as part of CryptoSwift. 

**Additional context**
https://en.bitcoin.it/wiki/Base58Check_encoding

**Comment by Sajjon** [comment]
There is an excellent implementation by [MartinR on StackOverflow](https://stackoverflow.com/questions/43091858/swift-hash-a-string-using-hash-hmac-with-ripemd160/43193583?noredirect=1#comment97958568_43193583), giving permission to use his code as a PR to this repo.

**Comment by krzyzanowskim** [comment]
@Sajjon want to make PR? with this code it should be straightforward

**Comment by kirilltitov** [comment]
RIPEMD320 would also be great!

**Comment by koolfreak** [comment]
also RipeMD256 also would be great!

**Reply by N-earning** [reply]
why MD256? what will give us as plus if not a secret?

**Comment by arnauddorgans** [comment]
Up !
